vic/vic_run/src/surface_fluxes.c

```diff
@@ -602,6 +593,11 @@ surface_fluxes(bool                 overstory,
                     INCLUDE_SNOW = false;
                 }
 
+                if (iter_snow.snow) {
+                    iter_aero_resist_veg[0] = iter_aero_resist_used[0];
+                    iter_aero_resist_veg[1] = iter_aero_resist_used[1];
+                }
```

Comment:
Because in the case of snow, aerodynamic resistance is modified in snow_intercept().  We need the "final" aerodynamic resistance over vegetation to be used in the weighted average in func_surf_energy_bal().

Now that I think of it, I wonder if I shouldn't just use the "veg" version for everything in the case of snow.  Perhaps I should add a case to func_surf_energy_bal() to cover this.  Either way, this updated value needs to be piped to func_surf_energy_bal() as I'm doing here.